src/mam4xx/convproc.hpp

```diff
@@ -356,16 +704,17 @@ void update_tendency_diagnostics(
 }
 // =========================================================================================
 // clang-format off
+template<class SubView, class ConstSubView>
```

Comment:
Would it make sense to consider a default for the second template parameter that just `const`ifies the first one?

Reply:
Ah, I suppose it doesn't matter if the signature is constructed to match the arguments. Though perhaps there's an opportunity to do some type checking.

Reply:
The View and ConstView template parameters are the difference between Kokkos::View<Real> and Kokkos::View<const Real> so it is non-trivial to derive the const from the non-const.  Some of these functions are templated just to avoid complicated declarations of Kokkos subviews of views. Since the values are assigned to an "auto" to begin with, making it a templated function means never having to write out the complicated Kokkos type.